### PR TITLE
Add k8s container

### PR DESCRIPTION
**Pull request by maeve-fpf**
## Status

Ready for review 

## Description of Changes

* Works towards issue https://github.com/freedomofpress/infrastructure/issues/855

This adds a container which runs Sphinx during the build, and serves the output with nginx. We will deploy this container in GKE to self-host the documentation and move off of RTD.

HTML output is relatively straightforward. There is an RTD-like theme in Sphinx. In order to get the functionality RTD gives us of a version switching widget, I added a small bit of template based on what RTD's JavaScript adds to the document. It's not as full-featured as RTD but it does show you if you're on `latest` or `stable` and lets you download the PDF.

PDF output required some wrangling: non-latin characters must be defined as LaTeX, and SVG inclusion does not work (it did not work on RTD either, however). In going through characters used in the source, I only fixed incorrectly encoded quotation marks, and did not try to make all the triangles used to denote submenu items consistent (there are U+25B6 ▶, U+25B8 ▸, and U+003E >).

## Testing

The output is deployed at https://staging-docs.securedrop.org/ (publicly accessible). To run locally, `docker build -f deploy/Dockerfile .`; and run with `docker run <imageid> -p 127.0.0.1:5080:5080`, then navigate to http://localhost:5080/. Because the container will be deployed behind another Nginx setup, it is set to omit port numbers from redirects, so you may have to add them back in.

The main redirect is from `/` to `/en/stable/`, to match the current behavior of RTD (as there is no localization the output is currently just copied to this directory to get the correct path). There are also redirects from the current RTD PDF URLs, `/_/downloads/en/stable/pdf/` (and `latest`), to the actual PDF files.

To verify that the content is different between stable and latest, you can check the "Set Up the Network Firewall" page - in latest, it has been split out into pfSense and OPNSense.

## Release 

Currently, k8s CI is set up to deploy a test branch to staging. Once this is merged, I can adjust that to run on pushes to `main` here.

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally - the container build does not run this, but we could add failing on that if desired.
- [ ] Doc link linting (`make docs-linkcheck`) passed - No, I'm currently getting a 500 from Gitter. Maybe an actual outage on their end?
- [x] You have previewed (`make docs`) docs at http://localhost:8000 - still works, although the version switcher/PDF link will be useless. There is no containerized live preview.

### CODE REVIEW

File: deploy/build
Comment (eloquence) [inline deploy/build:10]: In my testing, this works well to identify the correct version and respects post-release suffixes like `-1` and -2`. It will treat a tag like `-rc1` as being _newer_ than the release it refers to, but we currently don't use those in `securedrop-docs`.

**Comment by eloquence** [comment]
Thanks much for all the hard work on this @maeve-fpf. I was able to confirm that:

- [x] the documentation is built correctly in two versions, `stable` and `latest`
- [x]  these versions correspond to the `HEAD` of `main` and to the most recently tagged version
   - [x] respecting post-release modifiers: 2.2.0-1 > 2.2.0
   - [x] honoring SemVer: 2.2.21 > 2.2.3
- [x]  the URL paths correspond to the current documentation, i.e. no existing external links should break
- [x]  the PDF is generated (with errors as noted in the code comment) and corresponds mostly to the current PDF (see below)
- [x]  the menu selection for `stable` and `latest` is present _iff_ the documentation is built from a commit that includes the changes in this branch (I tested this by locally tagging this branch with a more recent version)
- [x]  the version selection works as expected

Differences of note:

1. The documentation no longer links directly to the GitHub URL corresponding to a specific section of the docs; the link points instead to the raw page source. IIRC we had previously discussed linking to a GitHub URL (could be the `HEAD` of `main` rather than the specific version the user is looking at). If that's easy enough to do, it seems somewhat more useful than the RST page source.
2. The PDF no longer includes a table of contents. That seems like a valuable addition given the length of the documentation.

IMO neither of these are blocking issues for this PR though, and we can investigate this further in follow-up. 

I'd appreciate a second pair of eyes from a maintainer before we merge (no need to repeat every single of the above checks though).

File: deploy/build
Comment (maeve-fpf) [reply deploy/build:10]: If we do end up needing to handle `-rc` versions, it looks like we can configure Git for that: https://git-scm.com/docs/git-config#Documentation/git-config.txt-versionsortsuffix

**Comment by maeve-fpf** [comment]
I've added a commit here, branched off of `2.1.0-2`, as a candidate to be tagged `2.1.0-3`. It contains a subset of these changes, squashed into one commit, so that a build with that tag checked out will generate a version switcher and PDF: https://github.com/freedomofpress/securedrop-docs/compare/2.1.0-2...stable-plus-k8s-container

**Review by cfm — CHANGES_REQUESTED** [review]
Following @eloquence's [review](https://github.com/freedomofpress/securedrop-docs/pull/295#issuecomment-1006283840), I've reviewed the diff and taken `staging-docs.securedrop.org` for a spin, testing especially:

- [x] the documentation is built correctly in two versions, `stable` and `latest`
- [x]  these versions correspond to the most recently tagged version and to the `HEAD` of `main` (respectively)
- [x]  the URL paths correspond to the current documentation, i.e. no existing external links should break
- [x]  the PDF is generated (with errors as noted in the code comment) and corresponds mostly to the current PDF (see below)
- [x]  the version selector widget works as expected

Inline I've requested (slash offered to make :-) one change in the underlying ReST source in this repository specifically (not a functional change to this implementation).

File: docs/development/updating_ossec.rst
Comment (maeve-fpf) [reply docs/development/updating_ossec.rst:145]: Thanks for the context! It looks, to me, like `ossec-logtest` is simply spitting back out the log line that was piped in, so if the input (from `echo`) is correctly encoded, I would expect the output to be too. I've added a comment linking to #199 -- does that seem good enough?

Comment (maeve-fpf) [reply docs/development/updating_ossec.rst]: The change is now reverted and we're going to handle rendering the incorrectly-encoded characters instead. If you want to add a comment to the source, please do!

**Comment by maeve-fpf** [comment]
Table of contents should now be working.

File: docs/development/updating_ossec.rst
Comment (cfm) [reply docs/development/updating_ossec.rst]: Thanks, @maeve-fpf, for out-of-band discussion and your follow-up on this point.  I've proposed an explanation in ad49ef8; feel free to revise as you see fit.

**Review by cfm — DISMISSED** [review]

**Comment by cfm** [comment]
Approved!—but I'll defer to you, @maeve-fpf and @eloquence, for merge, in light of any operational considerations I'm not in the loop on.

**Comment by maeve-fpf** [comment]
Merging now - noting here that for staging CI purposes, we are still building the "stable" docs off off this branch: https://github.com/freedomofpress/securedrop-docs/compare/2.1.0-2...stable-plus-k8s-container - tagging that commit as something `2.1.0-3` will remove the need for that. Let's observe if anything now in `main` causes problems for RTD, since we will still leave prod pointed at RTD for a bit.

**Review by eloquence — APPROVED** [review]
Approving and merging based on @cfm's and my prior review.

**Pull request merged**
